pandas/_libs/algos.pyx

```diff
@@ -1066,82 +1077,77 @@ def rank_1d(
                         out[lexsort_indexer[j]] = i - grp_start + 1
                 elif tiebreak == TIEBREAK_FIRST:
                     for j in range(i - dups + 1, i + 1):
-                        if ascending:
-                            out[lexsort_indexer[j]] = j + 1 - grp_start
-                        else:
-                            out[lexsort_indexer[j]] = 2 * i - j - dups + 2 - grp_start
+                        out[lexsort_indexer[j]] = j + 1 - grp_start
+                elif tiebreak == TIEBREAK_FIRST_DESCENDING:
+                    for j in range(i - dups + 1, i + 1):
+                        out[lexsort_indexer[j]] = 2 * i - j - dups + 2 - grp_start
```

Comment:
should it be obvious where `2 * i - j - dups + 2 - grp_start` comes from?  looks like equivalent to `(i - dups + 1) + (i - j + 1) - grp_start`

Reply:
Agreed, good opportunity to clarify here, have added some commentary